dotCMS/src/main/java/com/dotcms/rest/api/v1/contenttype/FieldResource.java

```diff
@@ -66,6 +62,45 @@ public Response createContentTypeField(@PathParam("typeId") final String typeId,
 		Response response = null;
 		
 		try {
+			final List<Field> fields = new JsonFieldTransformer(fieldsJson).asList();
+
+			for (Field field : fields) {
+				field = fapi.save(field, user);
+			}
+
+			List<Field> contentTypeFields = fapi.byContentTypeId(typeId);
+			response = Response.ok(new ResponseEntityView(new JsonFieldTransformer(contentTypeFields).mapList())).build();
+		} catch (DotStateException e) {
+
+			response = ExceptionMapperUtil.createResponse(null, "Field is not valid ("+ e.getMessage() +")");
+
+		} catch (NotFoundInDbException e) {
+
+			response = ExceptionMapperUtil.createResponse(e, Response.Status.NOT_FOUND);
+
+		} catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)